File: julia/bob/bob.jl
function bob(stimulus::AbstractString)
    function all_uppercase_not_empty(str)
        !isempty(str) && all(isuppercase, str)
    end

    stimulus = strip(stimulus)
    chars_only = replace(stimulus, r"[^a-zA-Z]" => "")
    if isempty(stimulus)
        "Fine. Be that way!"
    elseif '?' == stimulus[end]
        if all_uppercase_not_empty(chars_only)
            "Calm down, I know what I'm doing!"
        else
            "Sure."
        end
    elseif all_uppercase_not_empty(chars_only)
        "Whoa, chill out!"
    else
        "Whatever."
    end
end

File: nim/bob/bob.nim
import strutils, sequtils

func hey*(thing: string): string =
  if isEmptyOrWhiteSpace(thing):
    return "Fine. Be that way!"
  let sthing = thing.strip()
  let question = len(deduplicate(@sthing)) > 1 and sthing.endsWith("?")
  let yelling = sthing.any(isAlphaAscii) and
                sthing.allIt(not it.isAlphaAscii() or
                             it.isUpperAscii())
  if question and yelling:
    return "Calm down, I know what I'm doing!"
  if question:
    return "Sure."
  if yelling:
    return "Whoa, chill out!"
  return "Whatever."
